cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-integrating-aws.adoc

```diff
@@ -0,0 +1,391 @@
+----
+$ aws iam attach-role-policy --role-name "ostoy-sa-role" --policy-arn "${POLICY_ARN}"
+----
+
+== Creating the service account for your pod
+
+. Get the ARN for the AWS IAM role we created so that it will be included as an annotation when you create your service account by running the following command:
++
+[source,terminal]
+----
+$ export APP_IAM_ROLE_ARN=$(aws iam get-role --role-name=ostoy-sa-role --query Role.Arn --output text)
+----
++
+. Create the service account by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+name: ostoy-sa
+namespace: ${OSTOY_NAMESPACE}
+annotations:
+eks.amazonaws.com/role-arn: "$APP_IAM_ROLE_ARN"
+EOF
+----
++
+[IMPORTANT]
+====
+Do not change the name of the service account from "ostoy-sa" or you will have to change the trust relationship for the AWS IAM role.
+====
++
+. Grant the service account the `restricted` role by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user restricted system:serviceaccount:${OSTOY_NAMESPACE}:ostoy-sa
+----
++
+. Confirm that the previous step was successful by running the following command:
++
+[source,terminal]
+----
+$ oc describe serviceaccount ostoy-sa -n ${OSTOY_NAMESPACE}
+----
++
+.Example output
++
+[source,terminal]
+----
+Name:                ostoy-sa
+Namespace:           ostoy
+Labels:              <none>
+Annotations:         eks.amazonaws.com/role-arn: arn:aws:iam::000000000000:role/ostoy-sa-role
+Image pull secrets:  ostoy-sa-dockercfg-b2l94
+Mountable secrets:   ostoy-sa-dockercfg-b2l94
+Tokens:              ostoy-sa-token-jlc6d
+Events:              <none>
+----
+
+== Creating an S3 bucket
+
+. Create an S3 bucket using a manifest file by using one of the following options:
+** Run the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc apply -f -
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+name: ${OSTOY_NAMESPACE}-bucket
+namespace: ${OSTOY_NAMESPACE}
+spec:
+name: ${OSTOY_NAMESPACE}-bucket
+EOF
+----
++
+** Download the link:yaml/s3-bucket.yaml[command], then run it.
++
+[IMPORTANT]
+====
+The OSToy application expects to find a bucket named `<namespace\>-bucket`. If you use anything other than the namespace of your OSToy project, this feature will not work. For example, if our project is "ostoy", the value for `name` must be `ostoy-bucket`.
+====
++
+. Confirm the bucket was created by running the following command:
++
+[source,terminal]
+----
+$ aws s3 ls | grep ${OSTOY_NAMESPACE}-bucket
+----
+
+== Redeploying the OSToy app with the new service account
+
+. Run your pod with the service account you created.
+. Patch the `ostoy-frontend` deployment by running the following command:
++
+[source,terminal]
+----
+$ oc patch deploy ostoy-frontend -n ${OSTOY_NAMESPACE} --type=merge --patch '{"spec": {"template": {"spec":{"serviceAccount":"ostoy-sa"}}}}'
+----
++
+. Specify the service account:
```

Comment:
this is not an actual step.  It is just showing what the effect of the patch performed above is doing.